File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_list_response.py
"""Paginated envelope for GET /welcome-manuals."""
from __future__ import annotations

from platform_shared.schemas.pagination import ListResponse

from app.schemas.welcome_manuals.welcome_manual_summary import WelcomeManualSummary


class WelcomeManualListResponse(ListResponse[WelcomeManualSummary]):
    """Paginated response envelope for GET /welcome-manuals."""

File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-list-response.ts
import type { WelcomeManualSummary } from "./welcome-manual-summary";

/**
 * Paginated envelope returned by GET /welcome-manuals. Mirrors the shared
 * backend `ListResponse[WelcomeManualSummary]`.
 */
export interface WelcomeManualListResponse {
  items: WelcomeManualSummary[];
  total: number;
  has_more: boolean;
}
